src/ellipticcurve.rs

```diff
@@ -7,21 +7,20 @@ use num_bigint::{BigInt, BigUint};
 use std::fmt::Display;
 
 use crate::field::Field;
-use crate::ops::{AddRef, DivRef, MulRef, NegRef, ScMulRef, SubRef};
+use crate::ops::{AddRef, DivRef, MulRef, NegRef, ScMulRef, SubRef, Serialize, Deserialize};
 /// EcScalar models the behaviour of a scalar to multiply points.
-pub trait EcScalar: Display + AddRef + SubRef + MulRef + DivRef + NegRef {}
+pub trait EcScalar: Display + AddRef + SubRef + MulRef + DivRef + NegRef + Serialize {}
 
 /// EcPoint models the behaviour of a point on an elliptic curve.
-pub trait EcPoint<T>: Display + AddRef + SubRef + NegRef + ScMulRef<T>
+pub trait EcPoint<T>: Display + AddRef + SubRef + NegRef + ScMulRef<T> + Serialize
 where
     T: EcScalar,
 {
     fn is_zero(&self) -> bool;
-    fn serialize(&self, _: bool) -> Vec<u8>;
 }
 
 /// Curve trait allows to implement elliptic curve operations.
-pub trait EllipticCurve {
+pub trait EllipticCurve: Deserialize {
```

Comment:
I am not sure whether the **trait** must implement Deserialize.
I think each struct (Mont/Edw/Weier) must implement Deserialize.
You can also mimic the **FromFactory** trait in redox_ecc::field.

Reply:
I think the `Decode` trait should be enforced on the EllipticCurve object, as I think writing the points as bytes is a necessary piece of functionality. But if you think that it is not I can make it a trait on the structs.